aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -627,26 +886,31 @@ module aptos_framework::coin {
     public fun withdraw<CoinType>(
         account: &signer,
         amount: u64,
-    ): Coin<CoinType> acquires CoinStore {
+    ): Coin<CoinType> acquires CoinStore, CoinConversionMap, CoinInfo, PairedCoinType {
         let account_addr = signer::address_of(account);
-        assert!(
-            is_account_registered<CoinType>(account_addr),
-            error::not_found(ECOIN_STORE_NOT_PUBLISHED),
-        );
-
-        let coin_store = borrow_global_mut<CoinStore<CoinType>>(account_addr);
-        assert!(
-            !coin_store.frozen,
-            error::permission_denied(EFROZEN),
-        );
-
-        event::emit_event<WithdrawEvent>(
-            &mut coin_store.withdraw_events,
-            WithdrawEvent { amount },
-        );
-        event::emit(Withdraw<CoinType> { account: account_addr, amount });
-
-        extract(&mut coin_store.coin, amount)
+        if (is_account_registered<CoinType>(account_addr)) {
```

Comment:
what if it gets registered again?

Reply:
That's why I think it is hard to make this right. A simpler & foolproof solution is to consider both.

Reply:
I think writing this down rather than coding is where we need to be :P

Reply:
You are so responsive today. I am resolving those comment one by one.
Just talked to @alnoki and he is on my side. 
So I am going to write everything down in the AIP PR and then come back to refactor this to what's described in the AIP.

Reply:
> Just talked to @alnoki 

For clarify, posting here some of my thoughts from a conversation with @lightmark:

- IMO the average user shouldn't have to think about FA vs coin and if they are forced to there will ensure a public relations nightmare: if there are two forms of APT floating around in the wild then people will probably just get confused and therefore angry since the state of their holdings is unclear
- IMO The average end-user isn't equipped to make decisions about migrations, etc. and this abstraction that should definitely be hidden from them
- If FA and coin-style APT are treated as separate assets, then there could arise a situation where multiple DEXes or exchanges or quoting different prices for each type: in this case then what is the market price of APT? Such a discrepancy shouldn't happen IMO
- The idea of "not forcing users to migrate" is somewhat a inaccurate, because the fact that everything moving forward will be built on FA is itself a forcing function (analogy: saying users should be given the decision to migrate is like saying that the majority of Ethereum users weren't forced to accept the DAO hack rollback)
- Generally I endorse the on-the-fly conversions as of a04bd56 


Glad to move this conversation elsewhere if needed, but since I was tagged above I figured I'd capture those notes here